services/worker-manager/src/providers/azure.js

```diff
@@ -329,6 +501,84 @@ class AzureProvider extends Provider {
       throw error();
     }
 
+    // verify the subject of the signing certificate
+    const signerCommonName = crt.subject.getField({ name: 'commonName' });
+    if (!(signerCommonName &&
+          (signerCommonName.value === 'metadata.azure.com' ||
+           signerCommonName.value.endsWith('.metadata.azure.com')))) {
+      this.monitor.log.registrationErrorWarning({
+        message: 'Wrong PKCS#7 message signature subject',
+        error: `Expected "/CN=metadata.azure.com", got "${dnToString(crt.subject)}"`,
+      });
+      throw error();
+    }
+
+    // download the intermediate certificate if needed
+    let issuer = this.caStore.getIssuer(crt);
+    if (issuer === null) {
+      const authorityAccessInfo = getAuthorityAccessInfo(crt);
+      let method, location;
+      for (let i = 0; i < authorityAccessInfo.length; i++) {
+        method = authorityAccessInfo[i].method;
+        location = authorityAccessInfo[i].location;
+        if (method === 'CA Issuer' && location.startsWith('http:')) {
```

Comment:
I was surprised too, but it makes sense - if you a bootstrapping a certificate chain, you don't want to get into a situation where you need a different certificate to make the SSL connection to get a certificate...

[RFC 5280](https://tools.ietf.org/html/rfc5280#section-4.2.2.1) doesn't say HTTPS is forbidden, but does say 

> Conforming applications that support HTTP or FTP for accessing certificates MUST be able to accept individual DER encoded certificates and SHOULD be able to accept "certs-only" CMS messages.

and
 
> When the id-ad-caIssuers accessMethod is used, at least one instance SHOULD specify an accessLocation that is an HTTP [RFC2616] or LDAP [RFC4516] URI.

There are plenty of cases in the spec (FTP, LDAP, certs-only message instead of DER) that would be conforming but that we wouldn't parse. The fall-though logic will at least log the Authority Access Info so we can see what to add (or just download manually). Who knows, `https:` may show up some day...

Reply:
Let's hope we don't have to support LDAP :)